test/index.js

```diff
@@ -379,6 +381,54 @@ describe('Memory', function () {
         done();
     });
 
+    describe('stopping after set', function () {
```

Comment:
Let's do all of this in a single test. Also, set the test's `parallel: false`, as this isn't safe to run along other tests.